test_data/std/split_gl_vec_test.asm

```diff
@@ -0,0 +1,81 @@
+use std::machines::split::ByteCompare;
+use std::machines::split::split_gl::SplitGL;
+use std::machines::split::split_gl_vec::SplitGLVec8;
+use std::machines::large_field::memory::Memory;
+use std::machines::range::Byte2;
+
+let main_degree: int = 2**8;
+let memory_degree: int = 2**12;
+let split_degree: int = 2**10;
+let split_vec_degree: int = 2**10;
+
+machine Main with degree: main_degree {
+    reg pc[@pc];
+    reg X0[<=];
+    reg X1[<=];
+
+    Byte2 byte2;
+    Memory memory(byte2, memory_degree, memory_degree);
+
+    ByteCompare byte_compare;
+    SplitGL split_machine(byte_compare, split_degree, split_degree);
+
+    SplitGLVec8 split_vec_machine(memory, split_machine, split_vec_degree, split_vec_degree);
+
+    col fixed STEP(i) { 2 * i };
+
+    instr mstore X0, X1 ->
+        link ~> memory.mstore(X0, STEP, X1);
+
+    instr split X0, X1
+        link ~> split_vec_machine.split(X0, X1, STEP);
+
+    col witness val;
+    instr assert_eq X0, X1 ->
+        link ~> val = memory.mload(X0, STEP)
+    {
+        val = X1
+    }
+
+    function main {
+        // Store 8 field elements sequentially in memory
+        mstore 100, 0;
+        mstore 104, 0xffffffff00000000;
+        mstore 108, 0xfffffffeffffffff;
+        mstore 112, 0xabcdef0123456789;
+        mstore 116, 0x0000000100000002;
+        mstore 120, 0x0000000300000004;
+        mstore 124, 0x0000000500000006;
+        mstore 128, 0x0000000700000008;
+
+        // Split the previously stored field elements
+        split 100, 200;
+
+        // Assert the field elements are what was written
+        assert_eq 200, 0;
+        assert_eq 204, 0;
+
+        assert_eq 208, 0;
+        assert_eq 212, 0xffffffff;
+
+        assert_eq 216, 0xffffffff;
+        assert_eq 220, 0xfffffffe;
+
+        assert_eq 224, 0x23456789;
+        assert_eq 228, 0xabcdef01;
+
+        assert_eq 232, 0x00000002;
+        assert_eq 236, 0x00000001;
+
+        assert_eq 240, 0x00000004;
+        assert_eq 244, 0x00000003;
+
+        assert_eq 248, 0x00000006;
+        assert_eq 252, 0x00000005;
+
+        assert_eq 256, 0x00000008;
+        assert_eq 260, 0x00000007;
+
```

Comment:
Can you also add a repeated test case but where input_addr = output_addr to verify that the memory addresses and timestamps are working as intended

Reply:
Ok.